sdks/python/apache_beam/runners/interactive/display/pcoll_visualization_test.py

```diff
@@ -0,0 +1,152 @@
+    pv_1 = pv.PCollVisualization(self._pcoll)
+    pv_2 = pv.PCollVisualization(self._pcoll)
+    self.assertNotEqual(pv_1._dive_display_id, pv_2._dive_display_id)
+    self.assertNotEqual(pv_1._overview_display_id, pv_2._overview_display_id)
+    self.assertNotEqual(pv_1._df_display_id, pv_2._df_display_id)
+
+  @unittest.skipIf(sys.version_info < (3, 5, 3),
+                   'PCollVisualization is not supported on Python 2.')
+  @patch('apache_beam.runners.interactive.display.pcoll_visualization'
+         '.PCollVisualization._to_element_list', lambda x: [1, 2, 3])
+  def test_one_shot_visualization_not_return_handle(self):
+    self.assertIsNone(pv.visualize(self._pcoll))
+
+  def _mock_to_element_list(self):
+    yield [1, 2, 3]
+    yield [1, 2, 3, 4]
+    yield [1, 2, 3, 4, 5]
+    yield [1, 2, 3, 4, 5, 6]
+    yield [1, 2, 3, 4, 5, 6, 7]
+    yield [1, 2, 3, 4, 5, 6, 7, 8]
+
+  @unittest.skipIf(sys.version_info < (3, 5, 3),
+                   'PCollVisualization is not supported on Python 2.')
+  @patch('apache_beam.runners.interactive.display.pcoll_visualization'
+         '.PCollVisualization._to_element_list', _mock_to_element_list)
+  def test_dynamic_plotting_return_handle(self):
+    h = pv.visualize(self._pcoll, dynamic_plotting_interval=1)
+    self.assertIsInstance(h, timeloop.Timeloop)
+    h.stop()
+
+  @unittest.skipIf(sys.version_info < (3, 5, 3),
+                   'PCollVisualization is not supported on Python 2.')
+  @patch('apache_beam.runners.interactive.display.pcoll_visualization'
+         '.PCollVisualization._to_element_list', _mock_to_element_list)
+  @patch('apache_beam.runners.interactive.display.pcoll_visualization'
+         '.PCollVisualization.display_facets')
+  def test_dynamic_plotting_update_same_display(self,
+                                                mocked_display_facets):
+    fake_pipeline_result = runner.PipelineResult(runner.PipelineState.RUNNING)
+    ie.current_env().set_pipeline_result(self._p, fake_pipeline_result)
+    # Starts async dynamic plotting that never ends in this test.
+    h = pv.visualize(self._pcoll, dynamic_plotting_interval=0.001)
+    # Blocking so the above async task can execute at least 3 iterations.
+    timeout = time.time() + 0.1
+    while len(mocked_display_facets.call_args_list) < 3:
+      if time.time() > timeout:
+        break
+    # The first iteration doesn't provide updating_pv to display_facets.
+    _, first_kwargs = mocked_display_facets.call_args_list[0]
+    self.assertEqual(first_kwargs, {})
+    # The following iterations use the same updating_pv to display_facets and so
+    # on.
+    _, second_kwargs = mocked_display_facets.call_args_list[1]
+    updating_pv = second_kwargs['updating_pv']
+    for call in mocked_display_facets.call_args_list[2:]:
+      _, kwargs = call
+      self.assertIs(kwargs['updating_pv'], updating_pv)
+    h.stop()
+
+  @unittest.skipIf(sys.version_info < (3, 6),
```

Comment:
This is inconsistent with the rest of the tests. Perhaps make the rest of the tests 3.6+?

Reply:
The version is for testing code, not for code being tested.
The code being tested supports 3.5.3+
This specific test has assertion feature that was introduced in 3.6.

So the first 3 tests will be executed in both Py36 and Py37 test suites.
The last test will only be executed in Py37 suite.

Reply:
It would be useful to add your response as code comments here.

Reply:
Added